contracts/active-pool/src/state.rs

```diff
@@ -23,6 +16,17 @@ pub struct SudoParams {
     pub owner: Addr,
 }
 
+pub struct State<'a> {
```

Comment:
Hmm... I'm a bit confused about the usage of State. It needs the role provider address to instantiate state.roles (which has type RoleConsumer). But everywhere it's used, you're calling State::default which passes in a dummy "role_provider_address".

Shouldn't we be doing the following instead:
1) Store State as an item
```
pub const STATE: Item<State> = Item::new("state");
 ```
2. Creating new State object with the actual role_provider address in contract instantiation
```
   let role_provider =  deps.api.addr_validate(&msg.role_provider)?;
   let state = State {
        roles: RoleConsumer::new(role_provider),
    };
    STATE.save(deps.storage, &state)?;
```
TBH I'm a bit confused in general about the RoleConsumer usage. Wondering if there's a simpler way to accomplish the same thing

Reply:
@vernonjohnson `State` is a struct of storage objects that manages contract state. This pattern makes it easier to implement lifetimes for storage types.

`RoleConsumer` is struct that wraps `Item::new("role_provider_address")` and provides common functionality to query the role provider contract via its given address.